modules/executable-dependency/download-dependency-if-necessary.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python
+# A script to check if an executable is installed, and if not, download it. In either case, returns the path to the
+# executable. This script is meant to be executed from a Terraform external data source. To be as portable as possible,
+# this script has no external dependencies (i.e., nothing that doesn't come with Python), and should work with both
+# Python 2 and 3.
+
+
+import sys
+import distutils.spawn
+import json
+import tempfile
+import os
+import os.path
+import hashlib
+import platform
+import logging
+import errno
+
+try:
+    # Try the Python 3 import
+    from urllib.request import urlretrieve
+except ImportError:
+    # Fall back to the Python 2 immport
+    from urllib import urlretrieve
+
+
+DEFAULT_INSTALL_DIR_NAME = "download-dependency-if-necessary"
+
+
+logging.basicConfig(level=logging.INFO, format='%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+log = logging.getLogger(__name__)
+
+
+def main():
+    if len(sys.argv) != 5:
+        raise Exception("Usage: download-dependency-if-necessary.py EXECUTABLE DOWNLOAD_URL INSTALL_DIR APPEND_OS_ARCH")
+
+    _, executable, download_url, install_dir, append_os_arch_raw = sys.argv
```

Comment:
NIT: I recommend using [argparse](https://docs.python.org/2.7/library/argparse.html), since it's in stdlib

Reply:
It didn't seem to add much for parsing a few positional args... Not sure it's worth using for such a simple script?

Reply:
Looking again, I guess it creates a better CLI help text, helps with some of the arg parsing I'm doing manually, and will make it easier to extend the script in the future. OK, will change...

Reply:
Done: https://github.com/gruntwork-io/package-terraform-utilities/pull/28/commits/b81684ef86a69e472b1f2bd88fd31a2f766c4082